eZ/Publish/Core/MVC/Symfony/EventListener/SiteAccessMatchListener.php

```diff
@@ -117,4 +117,22 @@ private function getSiteAccessFromRequest(Request $request)
             )
         );
     }
+
+    /**
+     * @param Request $request
+     *
+     * @return string
+     */
+    private function getHost(Request $request)
+    {
+        if ($request->server->has('HTTP_X_ORIGINAL_ROUTE')) {
```

Comment:
is this something new in ezplatform?

Reply:
It is about https://github.com/ezsystems/ezpublish-kernel/pull/2452

Reply:
Still, it seems to me that this is a custom http header injected specifically by some environment. I'd rather see that environment fixed to provide the correct headers rather than ez doing special casing...
Long story: adding special treatment of custom http headers can easily lead to security breaches, as in every special http header should not be allowed to be sent by the original user, but only by the reverse proxy. And I am sure that no reverse proxy default config rewrites/deletes an incoming x-original-route header